src/message_filters/__init__.py

```diff
@@ -76,7 +76,10 @@ def __init__(self, *args, **kwargs):
         SimpleFilter.__init__(self)
         self.node = args[0]
         self.topic = args[2]
-        self.sub = self.node.create_subscription(*args[1:], self.callback, 10, **kwargs)
+        if kwargs.get('qos_profile'):
+            self.sub = self.node.create_subscription(*args[1:], self.callback, **kwargs)
+        else:
+            self.sub = self.node.create_subscription(*args[1:], self.callback, 10, **kwargs)
```

Comment:
You could mutate the `kwargs` rather than having `if`/`else` statements. Or at least, if you're going to do it this way, use the keyword before the `10`, I think it would be `qos_profile=10`.

Reply:
Agreed. I would do something like `kwargs.setdefault('qos_profile', 10)`.